File: platforms/ios/Dezel/Dezel/Date/DateParser.swift
/**
 * @class DateParser
 * @since 0.5.0
 * @hidden
 */
open class DateParser {

	//--------------------------------------------------------------------------
	// MARK: Properties
	//--------------------------------------------------------------------------

	/**
	 * @property parser
	 * @since 0.5.0
	 * @hidden
	 */
	private static var parser: DateFormatter = {
		let parser = DateFormatter()
		parser.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
		parser.timeZone = TimeZone.utc
		return parser
	}()

	//--------------------------------------------------------------------------
	// MARK: Methods
	//--------------------------------------------------------------------------

	/**
	 * @method parse
	 * @since 0.5.0
	 * @hidden
	 */
	public static func parse(_ string: String) -> Date {
		return self.parser.date(from: string) ?? Date()
	}	
}

File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/date/DateParser.kt
package ca.logaritm.dezel.date

import ca.logaritm.dezel.extension.Timezone
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*


/**
 * @class DateParser
 * @since 0.5.0
 * @hidden
 */
internal object DateParser {

	//--------------------------------------------------------------------------
	//  Properties
	//--------------------------------------------------------------------------

	/**
	 * @property parser
	 * @since 0.5.0
	 * @hidden
	 */
	private val parser: SimpleDateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US)

	//--------------------------------------------------------------------------
	//  Methods
	//--------------------------------------------------------------------------

	/**
	 * @constructor
	 * @since 0.5.0
	 */
	 init {
		this.parser.timeZone = Timezone.utc
	}

	/**
	 * Parses the specified ISO date string.
	 * @method parse
	 * @since 0.5.0
	 */
	public fun parse(string: String): Date {

		try {
			return this.parser.parse(string)
		} catch (e: ParseException) {
			return Date()
		}

	}
}
